framework/src/play/i18n/Messages.java

```diff
@@ -107,7 +107,7 @@ public static String getMessage(String locale, Object key, Object... args) {
         if (value == null && locale != null && locale.length() == 5 && locales.containsKey(locale.substring(0, 2))) {
             value = locales.get(locale.substring(0, 2)).getProperty(key.toString());
         }
-        if (value == null && defaults != null) {
+        if (value == null) {
```

Comment:
I think you should keep the test `default != null` as `defaults ` is public it can be set to null outside the class